doc/develop/test/twister.rst

```diff
@@ -759,3 +759,35 @@ dependencies between test cases.  For native_posix platforms, you can provide
 the seed to the random number generator by providing ``-seed=value`` as an
 argument to twister. See :ref:`Shuffling Test Sequence <ztest_shuffle>` for more
 details.
+
+Running test only mode
```

Comment:
Not a big fan of this approach. We should be able to support building on one host and running on another with less manual steps, supporting multiple boards and without too many modifications to flashers and runners...
IMO you are losing the connection between the original artefacts and the run results and moving prebuilt binaries  outside of the environment to load them somewhere else will make traceability and management of such environments very difficult.

Is it only about not copying everything produced in the build step? How about remove all build artefacts and just leave those needed for flashing and running? How are you going to support dealing with multiple boards in your solution and so on...?

Reply:
@nashif , you are right this approach is trying to decouple the build environments and running target. but leave user to choose to keep the traceabilities. .e.g they can wrapped remote build and remote run together in jenkins scripts.

> Is it only about not copying everything produced in the build step? How about remove all build artefacts and just leave those needed for flashing and running? How are you going to support dealing with multiple boards in your solution and so on...?

The current build artifect are linked with case name,  but actually what we need to run is only the flash /debugger  information. this can be once for all effort, instead of copy a build folder and delete many other contents.

To support multiply boards, we need use another process to do so. It is not a all in one feature, this feature is provide such as when we have a powful machine to build all binaries, and send each binary to runner, and runner can execute them one by one on given board. So we will need a jenkins like system to wrapper the steps and trigger different testplan for each application on each board.

Reply:
> another with less manual steps, supporting multiple boards and without too many modifications to flashers and runners...

the manual step lst here are one time job when you setting a board in run machine. and the modifications to flash and runners is just to remove the hard check on supportting files which dynamically linked with testcase name. instead using a static path can do the work, given the flash size and address is fixed for given board.

Reply:
I agree with @nashif. Looks quite fiddly. We already have `--runtime-artifact-cleanup` to leave only a bare minimum of build artifacts. However, I found it is not working if such artifacts are to be used later on with --test-only. I created a PR https://github.com/zephyrproject-rtos/zephyr/pull/46782  which expands the list of leftovers by a minimal set required to make flashing to work. Maybe such approach will fulfill your use case @hakehuang?

Reply:
> IMO you are losing the connection between the original artefacts and the run results and moving prebuilt binaries outside of the environment to load them somewhere else will make traceability and management of such environments very difficult.

@nashif , I will show the demo today, in test plan there is a runid, which is abided with the binary, so the tracebility is there.